src/aggregation/models/aggregationManager.ts

```diff
@@ -0,0 +1,90 @@
+import { InternalServerError, NotFoundError } from '@map-colonies/error-types';
+import type { Logger } from '@map-colonies/js-logger';
+import { inject, injectable } from 'tsyringe';
+import { SelectQueryBuilder } from 'typeorm';
+import { ConnectionManager } from '../../common/connectionManager';
+import { SERVICES } from '../../common/constants';
+import { CatalogClient } from '../../httpClient/catalogClient';
+import { PolygonPart } from '../../polygonParts/DAL/polygonPart';
+import { getEntitiesNames, isPolygonType } from '../../polygonParts/DAL/utils';
+import { PolygonPartsPayload } from '../../polygonParts/models/interfaces';
+import type { AggregationMetadata, AggregationParams } from './interfaces';
+
+@injectable()
+export class AggregationManager {
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.CONNECTION_MANAGER) private readonly connectionManager: ConnectionManager,
+    @inject(CatalogClient) private readonly catalogClient: CatalogClient
+  ) {}
+  public async getAggregationMetadata(aggregationParams: AggregationParams): Promise<AggregationMetadata> {
+    const logger = this.logger.child({ catalogId: aggregationParams.catalogId });
+    logger.info({ msg: 'metadata aggregation request' });
+
+    const findOptions = { id: aggregationParams.catalogId };
+    const layerMetadatas = await this.catalogClient.find(findOptions);
+    const layerMetadata = layerMetadatas.at(0);
+
+    if (layerMetadata === undefined) {
+      throw new NotFoundError('Could not find a catalog layer for the requested id');
+    }
+
+    if (layerMetadata.metadata?.productId === undefined) {
```

Comment:
Implement Zod schema for find at the service ATM and replace the fields validations

Reply:
added
we will add them from external source later